### PR TITLE
Drop support for older dependencies (see SPEC 0)

### PR DESCRIPTION
**Pull request by jarrodmillman**
https://scientific-python.org/specs/spec-0000/

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -17,15 +17,15 @@ jobs:
       MPLBACKEND: Agg
       CYTHON_TRACE: 1
       CYTHONSPEC: cython
-      NUMPY_MIN: numpy==1.22.4
+      NUMPY_MIN: numpy==1.23.0
       CYTHON_MIN: cython==0.29.35
```
Comment (jarrodmillman) [inline .github/workflows/tests.yml:21]:
```suggestion
      CYTHON_MIN: cython==3.0.4
```

Should this be kept in sync with pyproject.toml

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -129,23 +129,23 @@ jobs:
...
```
Comment (jarrodmillman) [inline .github/workflows/tests.yml:133]:
```suggestion
      CYTHON_MIN: cython==3.0.4
```

**Review by rgommers — APPROVED** [review]
Nice, this all LGTM. Especially the `fft`-related change; `numpy.fft` is indeed in much better shape now. And from numpy 2.0.0 onwards it is just as fast as SciPy, so when that becomes the minimum version it can be used unconditionally. Perhaps worth adding a comment to that effect?

I'd prefer to merge this right after creating the `1.6.x` release branch, to not make the NumPy 2.0 transition harder for projects that still want to keep Python 3.9 support.

File: pywt/_cwt.py [hunk 3]
```diff
@@ -14,27 +14,21 @@
...
-            """Round up size to the nearest power of two.
-
-            Given a number of samples `n`, returns the next power of two
-            following this number to take advantage of FFT speedup.
-            This fallback is less efficient than `scipy.fftpack.next_fast_len`
-            """
-            return 2**ceil(np.log2(n))
+    fftmodule = np.fft
+
+    # provide a fallback so scipy is an optional requirement
```
Comment (rgommers) [inline pywt/_cwt.py:23]:
```suggestion
    # provide a fallback so scipy is an optional requirement
    # note: numpy.fft in numpy 2.0 is as fast as scipy.fft, so could be used
    # unconditionally once the minimum supported numpy version is >=2.0
```

**Comment by rgommers** [comment]
In it goes - thanks Jarrod!

**Pull request merged**